packages/monaco/src/browser/monaco-editor-model.ts

```diff
@@ -86,8 +138,32 @@ export class MonacoEditorModel implements ITextEditorModel {
         return this.doSave(TextDocumentSaveReason.Manual);
     }
 
+    protected synchronizing = false;
+    async sync(): Promise<void> {
+        if (!this._dirty) {
+            const newText = await this.resource.readContents();
+            if (!this._dirty) {
+                const value = this.model.getValue();
+                if (value !== newText) {
+                    this.synchronizing = true;
+                    this.model.applyEdits([this.p2m.asTextEdit({
```

Comment:
What do you think, wouldn't it make sense to wrap into a try/catch/finally? So it will not get stuck in synchronizing state if an error happens.

Reply:
good point